internal/fullnode/pvc_builder.go

```diff
@@ -79,9 +85,23 @@ func pvcResources(crd *cosmosv1.CosmosFullNode) corev1.ResourceRequirements {
 		reqs = crd.Spec.VolumeClaimTemplate.Resources
 		size = reqs.Requests[corev1.ResourceStorage]
 	)
+
 	if autoScale := crd.Status.SelfHealing.PVCAutoScale; autoScale != nil {
-		if autoScale.RequestedSize.Cmp(size) > 0 {
-			reqs.Requests[corev1.ResourceStorage] = autoScale.RequestedSize
+		// This is the implementation using Int64 but it does not support fractions
+		//requestSize, unableToConvert := autoScale.RequestedSize.AsInt64()
+		//if unableToConvert == true {
+		//	fmt.Errorf("Unable to convert auto scale request")
+		//} else {
+		//	sizeWithPadding := resource.NewQuantity(int64(float64(requestSize)*SnapshotScalingFactor), resource.DecimalSI)
+		//	if sizeWithPadding.Cmp(size) > 0 {
+		//		reqs.Requests[corev1.ResourceStorage] = *sizeWithPadding
+		//	}
+		//}
+		requestedSize := autoScale.RequestedSize.DeepCopy()
+		newSize := requestedSize.AsDec()
```

Comment:
Not sure how safe it is to modify the requested size in place hence DeepCopy, not sure if the has the same semantics like cached values etc.

Reply:
Good call!